test/test_python_dispatch.py

```diff
@@ -523,5 +523,51 @@ def wrap(e):
         self.assertEqual(x.grad, y)
         self.assertEqual(y.grad, x)
 
+    def test_none_wrapping(self):
```

Comment:
Putting the discussion here so that it's easier to keep track of

> @zou3519
> 
> As a short term fix this seems OK?
> 
> I don't think this is a short term fix. The whole autograd uses undefined > Tensors for zero gradients. I don't think there is any plan to move that to optional any time soon.
Note that only backward function actually allow for this as any non-composite implicit function will fail if given an undefined Tensor.
> 
>  Note that this None <-> zeros gradient is already done in the custom autograd Function API where the two are the same.

@albanD I remember talking to someone (maybe Sebastian?) way back when and what they told me was that they wanted to excise undefined tensors in favor of `optional<Tensor>`. Our native_functions schema already does this. Do you think it's weird that a function like add can return a None tensor?

Reply:
> I remember talking to someone (maybe Sebastian?) way back when and what they told me was that they wanted to excise undefined tensors in favor of optional<Tensor>

Yes I think this is part of some grand BE plan. But I don't think anyone is ready to sign up to update all the autograd formulas and autograd engine to do this.
Especially if the current semantic we have is that from the python point of view, and empty optional looks the same as an undefined Tensor.

> Do you think it's weird that a function like add can return a None tensor?

I would agree with that. I use add as a simple example but it is actually really hard to trigger this (because we properly do conversions everywhere).
This came up when running backwards on subclass. Because when you delegate to the actual function, they can return undefined Tensors that get transformed to `None`. And so without this patch, delegation like the one done for LoggingTensor in [here](https://github.com/pytorch/pytorch/blob/c5e5264be26fc84ada3c63d7add579ab19d4834c/test/test_python_dispatch.py#L64) is failing. It works seamlessly with this patch.